### PR TITLE
Documentation: Add additional terminologies in the glossary.md

### PR DESCRIPTION
**Pull request by leemgs**
It is trivial patch. However, it will be helpful to developers that
learn the internals of CoreCLR via the source code and documents.

Signed-off-by: Geunsik Lim <geunsik.lim@samsung.com>

### CODE REVIEW

File: Documentation/project-docs/glossary.md [hunk 1]
```diff
@@ -5,13 +5,20 @@ This glossary defines terms, both common and more niche, that are important to u
 
 As much as possible, we should link to the most authoritative and recent source of information for a term. That approach should be the most helpful for people who want to learn more about a topic.
 
-* CLR: Common Language Runtime
+* BOTR: Book Of The Runtime.
+* CLR: Common Language Runtime.
 * COMPlus: An early name for the .NET platform, back when it was envisioned as a successor to the COM platform (hence, "COM+"). Used in various places in the CLR infrastructure, most prominently as a common prefix for the names of internal configuration settings. Note that this is different from the product that eventually ended up being named [COM+](https://msdn.microsoft.com/en-us/library/windows/desktop/ms685978.aspx).
 * COR: [Common Object Runtime](http://www.danielmoth.com/Blog/mscorlibdll.aspx). The name of .NET before it was named .NET.
+* CORECLR: CORE Common Language Runtime.
```
Comment (adityamandaleeka) [inline Documentation/project-docs/glossary.md]: Casing should be 'CoreCLR' (also, I'm not really sure this definition is helpful).

File: Documentation/project-docs/glossary.md [hunk 2]
```diff
@@ -5,13 +5,20 @@ This glossary defines terms, both common and more niche, that are important to u
...
 * DAC: Data Access Component. An abstraction layer over the internal structures in the runtime.
 * EE: Execution Engine.
+* GC: [Garbage Collector](https://github.com/dotnet/coreclr/blob/master/Documentation/botr/garbage-collection.md).
+* JIT: [Just-in-Time](https://github.com/dotnet/coreclr/blob/master/Documentation/botr/ryujit-overview.md) compiler. RyuJIT is the code name for the next generation Just-in-Time(aka "JIT") for the .NET runtime.
 * LCG: Lightweight Code Generation. An early name for [dynamic methods](https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/Reflection/Emit/DynamicMethod.cs).
+* MSCORLIB: MicroSoft CORe LIBrary. It consists of mscorlib.dll and System.Private.CoreLib.dll.
```
Comment (adityamandaleeka) [inline Documentation/project-docs/glossary.md]: 'mscorlib'. Also, I believe the 'COR' comes from 'Common Object Runtime'.

File: Documentation/project-docs/glossary.md [hunk 3]
```diff
@@ -5,13 +5,20 @@ This glossary defines terms, both common and more niche, that are important to u
...
+* NIG: Native Image Generator.
```
Comment (adityamandaleeka) [inline Documentation/project-docs/glossary.md]: This should be NGen. I've never seen this other acronym used to refer to it. Did you find this somewhere in the code or docs?

File: Documentation/project-docs/glossary.md (hunk 1 above)
Comment (leemgs) [inline Documentation/project-docs/glossary.md]: In this PR, I will remove "CORECLR" that you commented.

File: Documentation/project-docs/glossary.md (hunk 2 above)
Comment (leemgs) [inline Documentation/project-docs/glossary.md]: I will also remove "CORECLR" that you commented in this PR.

File: Documentation/project-docs/glossary.md (hunk 3 above)
Comment (leemgs) [inline Documentation/project-docs/glossary.md]: I have saw "NIG" from the some files. It seems that "NGen" is better than "NIG" in glossary.md file.

**Comment by leemgs** [comment]
@adityamandaleeka , Thank you for commenting the valuable feedback. Updated.

**Comment by jkotas** [comment]
LGTM. Thanks!

**Pull request merged**